### PR TITLE
CentOS 7, Ansible and the end of Python 2

### PR DESCRIPTION
**Pull request by volcan01010**
### Blog Post Title

* CentOS 7, Ansible and the end of Python 2

This post summarises Declan and I's findings after we spend lots of this week fixing our Ansible / pipeline issue.  We didn't find a single compilation of information about these problems online, so this post should be useful for others outside BGS as well as providing a reference for whenever we hit these problems again.

### Link to rendered version

Please include a link to the rendered version of your blog post so it can be seen its intended display format,

* https://volcan01010.github.io/BritishGeologicalSurvey.github.io/devops/centos7-python2-end-of-life/

### CODE REVIEW

File: _posts/2020-02-21-centos7-python2-end-of-life.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+a few of our long-used jobs recently began to fail as a result Python 2 / Python 3 mismatches.
+Officially, CentOS 7 is [supported until
+June 2024](https://wiki.centos.org/About/Product), but I think the end of
+Python 2 will hasten its demise.
+
+The notes below explain how we solved our issues with CentOS 7 and Ansible.
+Hopefully they will be helpful to others.
+
+
+## Background
```
Comment (dvalters) [inline _posts/2020-02-21-centos7-python2-end-of-life.md:28]: Might be a template issue, but on my browser this renders as "BackgroundPermalink" (with the link under the Permalink text)

File: _posts/2020-02-21-centos7-python2-end-of-life.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+categories:
+  - devops
+tags:
+  - Python
+  - CentOS
+  - Ansible
+  - continuous-integration
+---
+
+The [sunsetting of Python 2](https://www.python.org/doc/sunset-python-2/) on 1 January 2020 has been causing problems our CentOS 7 server administration.
```
Comment (dvalters) [inline _posts/2020-02-21-centos7-python2-end-of-life.md]: causing problems ^with^ our CentOS 7 server

File: _posts/2020-02-21-centos7-python2-end-of-life.md (hunk 1 above)
Comment (dvalters) [reply _posts/2020-02-21-centos7-python2-end-of-life.md:28]: And all the other headings and subheadings

**Review by dvalters — COMMENTED** [review]
I guess we could add in an explanation of why we believe Python 3 will actually hasten the demise of Centos 7 (I can add a few words?) (as per Wayne's suggestion on teams.)

**Review by dvalters — APPROVED** [review]

**Comment by dvalters** [comment]
Looks good now - the link text seems to have resolved itself...

**Review by pdbebgs — APPROVED** [review]

**Pull request merged**